op-node/rollup/derive/attributes.go

```diff
@@ -122,5 +130,6 @@ func (ba *FetchingAttributesBuilder) PreparePayloadAttributes(ctx context.Contex
 		NoTxPool:              true,
 		GasLimit:              (*eth.Uint64Quantity)(&sysConfig.GasLimit),
```

Comment:
Do the `EcotoneNetworkUpgradeTransactions` consume a deterministic amount of gas, or is there at least a deterministic upper bound? If yes, we could add this amount to the `GasLimit` to ensure user deposit transactions have the same amount of gas available in the upgrade block.

We could let `EcotoneNetworkUpgradeTransactions` return this gas limit above and do something like

```go
gaslimit := (eth.Uint64Quantity)(sysConfig.GasLimit)

    // on line 104
    upgradeTxs, upgradeGas, err = EcotoneNetworkUpgradeTransactions()
    gaslimit += upgradeGas
```
before and then here
```suggestion
		GasLimit:              &gaslimit,
```
So in this sense, we would build up the `gaslimit` value just as we build up the `txs`.

Reply:
This is a good idea, I believe that this needs to be specified better. They should consume a deterministic amount of gas cc @danyalprout 

The spec says TBC for the gas values of the individual txs, this is under specified still

Reply:
Actually I don't think we want to modify the gas limit here, it may break things. Not sure how this interacts with 1559. Thoughts @protolambda @trianglesphere @ajsutton ?

Reply:
I'll add a todo issue to review the gas-limits of the other upgrade txs. The gas-limit of the only L2-Dencun-specific tx is a fixed constant, defined in the EIP, and very low.

Reply:
Added to https://github.com/ethereum-optimism/protocol-quest/issues/62 description